### ISSUE TITLE
[bug] @import'ed tailwind CSS does not invalidate the cache after error with @layer

### ISSUE THREAD

**Issue by talss89** [roots/bud#2455]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

When Bud enters an error state after, say, a syntax error, it fails to rebuild and exit the error state.

This only happens when using `@roots/bud-tailwindcss` and in the following circumstances:

- The CSS file throwing the TWCSS error must be `@import`'ed from the entrypoint (ie. >1 module deep)
- The error must be thrown from within a `@layer` statement

Once bud 'goes red' and enters the error state, it appears to stop watching the CSS file that threw the error. Fixing the error, then quitting and re-running bud builds fine.

`--force` seems to have no effect, not that we should really need it.

### Expected Behavior

Once the error has been resolved, bud rebuilds and the exits the error state.

### Actual Behavior

Bud stays locked in the error state.

### Steps To Reproduce

Here's a repo with a minimal example: https://github.com/talss89/tailwindcss-layer-cache-invalidation-bug

Check out `./src/child.css` - run `yarn dev`, then uncomment the garbage and save.

---

Here's how to replicate outside of that repo:

1. `npx create-bud-app` - Select tailwindcss
2. Create a child CSS file, and `@import` it into `index.css`
3. Inside the child CSS file:

```css
@layer base {
  h1 {
    @apply text-xl;
  }
}
```

4. Run `bud dev`
5. Change the child CSS file to cause a Tailwind error; ie:

```css
@layer base {
  h1 {
    @apply hfjkdgfg;
  }
}
```

6. Bud will then report the error
7. Revert the child CSS file back to the original
8. Bud remains in the error state

### version

6.16.1

**Comment by kellymears** [comment]
I can reproduce it.

Not sure what causes it or what we can do to address it. It has some qualities that make me wonder if it is an upstream issue.

One important thing to note is that the syntax error is actually reported in `./node_modules/tailwindcss/[layer name].css`, not the module that contains the error.   

```ts
╭ tailwindcss [cba9ce73a58c9a3e]                                        ./dist
│
│ │ Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
│ │ SyntaxError
│ │ (1:1)
│ │ ./node_modules/tailwindcss/base.css The
│ │  `foo` class does not exist. If `foo` is a custom class, make sure it is
│ │ defined within a `@layer` directive.
│ │ > 1 | @tailwind base;
│ │     | ^
│ │   2 |
│
╰ 1 error
```

When we remove the error postcss doesn't re-evaluate `./node_modules/tailwindcss/base.css`, which makes sense as we obviously didn't change that module. Sure enough, if you open `./node_modules/tailwindcss/base.css` manually and save it then normal behavior is restored.

### Workaround

Save the entrypoint css module (even if there have been no changes), and this also gets things back on track again.

### Maybe better workaround 

Create a separate module for each layer you wish to modify, and import them in the entrypoint:

```css
// entrypoint.css
@import 'base';

@tailwind components;
@tailwind utilities;
```

```css
// base.css
@tailwind base;

@layer base {
  #root {
    @apply bg-gray-400;
    @apply h-screen;
    @apply w-screen;
    @apply relative;
  }
}
```

If you make a syntax error in `base.css` the error originates from `./src/base.css` (rather than `./node_modules/tailwindcss/base.css`), as should be expected. If you fix the syntax error everything works as expected.

I like this because it keeps the layers cleanly organized while avoiding this issue.

### Next steps

I'm not sure what the next steps are for bud.js or what we should be doing differently, if anything. I'll have to explore this more.

**Comment by talss89** [comment]
Thanks for looking into this so quickly.

I've had a snoop around the Tailwind docs and src, and three things are apparent:

1. The docs no longer mention my `@import 'tailwindcss/base';` syntax.
2. [Tailwind looks for `@tailwind` directives when invalidating the cache](https://github.com/tailwindlabs/tailwindcss/blob/0e7f93699556f92c4d949b0cb0cfacb76169ff5a/src/lib/cacheInvalidation.js#L39C1-L43C4)
3. I'm doing it all wrong

So, in summary, I'm not sure anything needs to change here. Your 2nd workaround is actually the correct way of doing things. I'll close this.

[Looks like Sage suggests the `@import` syntax out of the box](https://github.com/roots/sage/blob/main/resources/styles/app.css), so might raise a PR over there.
